tools/libvirt_test.sh

```diff
@@ -82,6 +82,15 @@ BLUEPRINT_FILE=${TEMPDIR}/blueprint.toml
 COMPOSE_START=${TEMPDIR}/compose-start-${IMAGE_KEY}.json
 COMPOSE_INFO=${TEMPDIR}/compose-info-${IMAGE_KEY}.json
 
+# Create SSH key
+SSH_KEY=/tmp/id_rsa
+ssh-keygen -f "${SSH_KEY}" -N "" -q
+
+# Change cloud-init/user-data ssh key
+key=" -ssh-rsa $(sed "1d;38d;s/\n//g" "${SSH_KEY}"| tr -d '[:space:]')"
+sed -i "/ssh-rsa/d" /usr/share/tests/osbuild-composer/cloud-init/user-data # In case there's a key for some reason
+echo "$key" >> /usr/share/tests/osbuild-composer/cloud-init/user-data
```

Comment:
That is just going to append the key at the end of the file which won't work. You need to put it under the `ssh_authorized_keys:` part of `user-data` file... 
I suggest moving the whole user part at the end and just amend it with the key ( I expect cloud-init doesn't care about the order of things there as long as the syntax is correct). Also don't forget the correct indent (adding spaces before it)